diff --git a/arch/arm/boot/dts/sun8i-h2-plus-orangepi-zero.dts b/arch/arm/boot/dts/sun8i-h2-plus-orangepi-zero.dts
index 3b6d99ae4..5094a3a08 100644
--- a/arch/arm/boot/dts/sun8i-h2-plus-orangepi-zero.dts
+++ b/arch/arm/boot/dts/sun8i-h2-plus-orangepi-zero.dts
@@ -208,6 +208,10 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart2_pins>;
 	status = "okay";
+	linux,rs485-enabled-at-boot-time;
+	rs485-rts-active-low;
+	rts-gpios = <&pio 0 3 1>;
+	rs485-rts-delay = <0 0>;
 };
 
 &usb_otg {
@@ -231,7 +235,11 @@
                pins = "PA3";
                function = "uart1";
        };
-
+	uart2_rts_cts: uart2_rts_cts
+	{
+		pins = "PA3";
+		function = "uart2";
+	};
 	wifi_wake: wifi_wake {
 		pins = "PG10";
 		function = "gpio_in";
